internal/services/nginx/nginx_certificate_data_source.go

```diff
@@ -0,0 +1,118 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type CertificateDataSourceModel struct {
+	Name                   string `tfschema:"name"`
+	NginxDeploymentId      string `tfschema:"nginx_deployment_id"`
+	KeyVirtualPath         string `tfschema:"key_virtual_path"`
+	CertificateVirtualPath string `tfschema:"certificate_virtual_path"`
+	KeyVaultSecretId       string `tfschema:"key_vault_secret_id"`
+}
+
+type CertificateDataSource struct{}
+
+var _ sdk.DataSource = CertificateDataSource{}
+
+func (m CertificateDataSource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"nginx_deployment_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: nginxdeployment.ValidateNginxDeploymentID,
+		},
+	}
+}
+
+func (m CertificateDataSource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"key_virtual_path": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"certificate_virtual_path": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"key_vault_secret_id": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+	}
+}
+
+func (m CertificateDataSource) ModelObject() interface{} {
+	return &CertificateDataSourceModel{}
+}
+
+func (m CertificateDataSource) ResourceType() string {
+	return "azurerm_nginx_certificate"
+}
+
+func (m CertificateDataSource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Nginx.NginxCertificate
+			var model CertificateDataSourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return err
+			}
+			deploymentId, err := nginxdeployment.ParseNginxDeploymentID(model.NginxDeploymentId)
+			if err != nil {
+				return fmt.Errorf("error parsing NGINX deployment ID %s: %+v", deploymentId, err)
+			}
+			id := nginxcertificate.NewCertificateID(
+				deploymentId.SubscriptionId,
+				deploymentId.ResourceGroupName,
+				deploymentId.NginxDeploymentName,
+				model.Name,
+			)
+			result, err := client.CertificatesGet(ctx, id)
+			if err != nil {
+				if response.WasNotFound(result.HttpResponse) {
+					return fmt.Errorf("%s was not found", id)
+				}
+				return fmt.Errorf("reading %s: %+v", id, err)
+			}
+
+			output := CertificateDataSourceModel{
+				Name:              id.CertificateName,
+				NginxDeploymentId: deploymentId.ID(),
+			}
+
+			if model := result.Model; model != nil {
+				prop := result.Model.Properties
+				output.KeyVirtualPath = pointer.ToString(prop.KeyVirtualPath)
+				output.KeyVaultSecretId = pointer.ToString(prop.KeyVaultSecretId)
+				output.CertificateVirtualPath = pointer.ToString(prop.CertificateVirtualPath)
```

Comment:
We should be using the generic `pointer.From` here and not `.ToString`
```suggestion
				output.KeyVirtualPath = pointer.From(prop.KeyVirtualPath)
				output.KeyVaultSecretId = pointer.From(prop.KeyVaultSecretId)
				output.CertificateVirtualPath = pointer.From(prop.CertificateVirtualPath)
```